corehq/apps/data_interfaces/views.py

```diff
@@ -540,3 +550,261 @@ def xform_management_job_poll(request, domain, download_id,
+            )
+
+    def create_rule(self):
+        with transaction.atomic():
+            rule = AutomaticUpdateRule.objects.create(
+                domain=self.domain,
+                name=self.rule_form.cleaned_data['name'],
+                case_type=self.rule_form.cleaned_data['case_type'],
+                active=True,
+                server_modified_boundary=self.rule_form.cleaned_data['server_modified_boundary'],
+            )
+            self.create_criteria(rule)
+            self.create_actions(rule)
+
+    def post(self, request, *args, **kwargs):
+        if self.rule_form.is_valid():
+            self.create_rule()
+            return HttpResponseRedirect(reverse(AutomaticUpdateRuleListView.urlname, args=[self.domain]))
+        return self.get(request, *args, **kwargs)
+
+
+class EditAutomaticUpdateRuleView(AddAutomaticUpdateRuleView):
+    urlname = 'edit_automatic_update_rule'
+    page_title = ugettext_lazy("Edit Automatic Update Rule")
+
+    @property
+    @memoized
+    def rule_id(self):
+        return self.kwargs.get('rule_id')
+
+    @property
+    def page_url(self):
+        return reverse(self.urlname, args=[self.domain, self.rule_id])
+
+    @property
+    @memoized
+    def rule(self):
+        try:
+            rule = AutomaticUpdateRule.objects.get(pk=self.rule_id)
+        except AutomaticUpdateRule.DoesNotExist:
+            raise Http404()
+
+        if rule.domain != self.domain or rule.deleted:
+            raise Http404()
+
+        return rule
+
+    @property
+    def initial_rule_form(self):
+        conditions = []
+        for criterion in self.rule.automaticupdaterulecriteria_set.all():
+            conditions.append({
+                'property_name': criterion.property_name,
+                'property_match_type': criterion.match_type,
+                'property_value': criterion.property_value,
+            })
+
+        update_case = False
+        update_property_name = None
+        update_property_value = None
+        for action in self.rule.automaticupdateaction_set.all():
+            if action.action == AutomaticUpdateAction.ACTION_UPDATE:
+                update_case = True
+                update_property_name = action.property_name
+                update_property_value = action.property_value
+                break
+
+        initial = {
+            'name': self.rule.name,
+            'case_type': self.rule.case_type,
+            'server_modified_boundary': self.rule.server_modified_boundary,
+            'conditions': json.dumps(conditions),
+            'update_case': update_case,
+            'update_property_name': update_property_name,
+            'update_property_value': update_property_value,
+        }
+        return AddAutomaticCaseUpdateRuleForm(initial=initial)
+
+    def post(self, request, *args, **kwargs):
+        if self.rule_form.is_valid():
+            self.rule.soft_delete()
```

Comment:
Every time we edit a rule, we soft delete the old one (by setting a `deleted` flag).  I figured it was a good idea to always have the complete history of rules that existed in case we ever need to debug why a case update happened (and since this table will be tiny anyway).  But I'm happy to get rid of the version history if there are strong feelings against it.